gpii/node_modules/flowManager/src/UserLogonStateChange.js

```diff
@@ -115,49 +142,82 @@ fluid.defaults("gpii.flowManager.userLogonStateChange.matchMakingStateChangeHand
 });
 
 
-// Incomplete grade - local FlowManager adds "gpii.flowManager.userLogonStateChange.matchMakingStateChangeHandler",
+// Incomplete grade - local FlowManager adds "gpii.flowManager.userLogonHandling.matchMakingStateChangeHandler",
 // whereas untrusted adds "gpii.flowManager.untrusted.stateChangeHandler"
 
-fluid.defaults("gpii.flowManager.userLogonStateChange.handler", {
+fluid.defaults("gpii.flowManager.proximityTrigger.handler", {
     gradeNames: ["kettle.request.http"],
     invokers: {
         handleRequest: {
-            funcName: "gpii.flowManager.userLogonStateChange.handleRequest",
-            args: ["{request}.req.params.userToken", "{flowManager}.lifecycleManager", "{that}"]
+            funcName: "gpii.flowManager.userLogonHandling.handleProximityTriggerRequest",
+            args: ["{that}", "{flowManager}.lifecycleManager", "{request}.req.params.userToken"]
         }
     }
 });
 
+/*
+ * Function to check whether a proximity trigger should bounce
+ *
+ * @param lifecycleManager {Object}: the lifecycleManager module
+ * @param debounceTimeMs {int}: the time in milliseconds for which to debounce after last proximityTrigger was fired
+ * @param userToken {string}: The token of the user for whom the proximityTrigger was fired
+ * @return {boolean}: Returns true if the proximityTrigger should be ignored (i.e. is considered a bounce). This is done according
+ *              to the following rule: any RFID actions is ignored for <mytoken> if a login/logout for <mytoken> is in progress
+ *              OR if the last login/logout process for <mytoken> finished less than {debounceTimeMs} milliseconds ago
+ */
+gpii.flowManager.userLogonHandling.isBounce = function (lifecycleManager, debounceTimeMs, userToken) {
+    // if login/logout is in progress for <userToken> we consider it a bounce
+    if (lifecycleManager.model.logonChange.inProgress && lifecycleManager.model.logonChange.userToken === userToken) {
```

Comment:
Nice, reusable function - body is factored a little oddly. I recommend expressing it like this:

var now = Date.now(), model = lifecycleManager.model;
if (userToken === model.logonChange.userToken) {
    return model.logonChange.inProgress || now - model.logonChange.timeStamp < debounceTimeMs;
} else {
    return false;
}